### ISSUE TITLE
Custom Loading Bar

### ISSUE THREAD

**Issue by Bhupesh-V** [Bhupesh-V/defe#3]
A nice addition to the CLI would be for e.g a different colour of loading bar or emojis in progress bar

**Comment by thePlebDev** [comment]
is this still open?

**Comment by Bhupesh-V** [comment]
> is this still open?

Yes, it is.
Make sure to discuss here about your ideas on how to do this

**Comment by jueyaye** [comment]
I had a quick look through your project and it looks like you use [tqdm](https://github.com/tqdm/tqdm) for your progress bars currently:
https://github.com/Bhupesh-V/defe/blob/master/core/feedcore.py#L7
Going through the code for tqdm it seems liker there's an [option to specify a colour](https://github.com/tqdm/tqdm/blob/00dabd63d759a1cac5a32c86aa773ea68e8d8ddc/tqdm/std.py#L947-L948) when you call `tqdm()` here:
https://github.com/Bhupesh-V/defe/blob/master/core/feedcore.py#L43-L49

Depending on how you would like to implement it. I imagine it would be pretty simple to add functionality which would allow users to set the colours of the progress bar as a separate CLI command or as an argument for your current commands. Which sounds best to you?

**Comment by Bhupesh-V** [comment]
> I had a quick look through your project and it looks like you use [tqdm](https://github.com/tqdm/tqdm) for your progress bars currently:
> https://github.com/Bhupesh-V/defe/blob/master/core/feedcore.py#L7
> Going through the code for tqdm it seems like there's an [option to specify a color](https://github.com/tqdm/tqdm/blob/00dabd63d759a1cac5a32c86aa773ea68e8d8ddc/tqdm/std.py#L947-L948) when you call `tqdm()` here:
> https://github.com/Bhupesh-V/defe/blob/master/core/feedcore.py#L43-L49
>
> Depending on how you would like to implement it. I imagine it would be pretty simple to add functionality that would allow users to set the colors of the progress bar as part of a config file or as an argument for your current commands. Which sounds best to you?
oo

That sounds great although I don't think we need to make it accessible to users to change colors.
Right now we can just maybe randomize colors on every feed lookup from an already set of defined colors.

What do you think?

**Comment by jueyaye** [comment]
Yeah, very doable. I'll try and write up a PR for it. Not sure how you would do loading bar with emojis but maybe there are some other options using `tqdm()`

**Comment by jueyaye** [comment]
Looks like we can use this option to customise the character used to complete the fill the meter. 
https://github.com/tqdm/tqdm/blob/00dabd63d759a1cac5a32c86aa773ea68e8d8ddc/tqdm/std.py#L888-L890
I did a little searching and found this package which it makes easy to just insert emojis.
https://pypi.org/project/emoji/
I did a quick test with the 👍 emoji and this is the result:
![image](https://user-images.githubusercontent.com/11943789/96359747-a43aa380-1159-11eb-84ba-cbb9bc9a5cca.png)
Does this look like what you were after?

**Comment by Bhupesh-V** [comment]
> Looks like we can use this option to customise the character used to complete the fill the meter.
> https://github.com/tqdm/tqdm/blob/00dabd63d759a1cac5a32c86aa773ea68e8d8ddc/tqdm/std.py#L888-L890
> I did a quick test with the  emoji and this is the result:
> ![image](https://user-images.githubusercontent.com/11943789/96359747-a43aa380-1159-11eb-84ba-cbb9bc9a5cca.png)
> Does this look like what you were after?

This exactly looks what I needed although thumbs up might annoy people 😅,  we can use 📥 or 📩 to display progress what do you think?
Something like this, maybe
![cli-color](https://d399irh3pgqnz3.cloudfront.net/prod/imagine/blog_image/uploads/blog/fun-with-symfonys-console.png)

**Comment by jueyaye** [comment]
Ah I see, yeah definitely. I believe, that ascii option updates what's filled (usually smooth blocks), not sure if it there is a way to cleanly have it so only the last option is the emoji but I'll endeavour to find out.
